dev/com.ibm.ws.classloading.bells/src/com/ibm/ws/classloading/bells/internal/Bell.java

```diff
@@ -205,41 +209,62 @@ public void removedService(ServiceReference<Library> libraryRef, List<ServiceReg
         }
     }
 
-    boolean getSpiVisibility(Boolean enableSpiVisibilityAttr, String libraryId) {
-        final boolean enableSpiVisibility = (enableSpiVisibilityAttr == null) ? false : enableSpiVisibilityAttr.booleanValue();
-        if (enableSpiVisibility) {
-            if (GLOBAL_SHARED_LIBRARY_ID.equals(libraryId)) {
-                // The liberty "global" library is intended for use by Jakarta EE applications, not OSGi services
-                // TODO: should we establish this restriction for all BELL configurations?
-                Tr.warning(tc, "bell.spi.visibility.disabled.libref.global");
-            } else {
-                Tr.info(tc, "bell.spi.visibility.enabled", libraryId);
-                return true;
-            }
+    private Set<String> getServiceNames(String[] configuredServices) {
+        if (configuredServices == null || configuredServices.length == 0) {
+            return Collections.emptySet();
         }
-        return false;
+        return Collections.unmodifiableSet(new HashSet<String>(Arrays.asList(configuredServices)));
     }
 
     private static boolean issuedBetaMessage = false;
 
     private boolean betaFenceCheck() {
-        boolean isBeta = com.ibm.ws.kernel.productinfo.ProductInfo.getBetaEdition();
-        if (!issuedBetaMessage) {
-            if (!isBeta) {
-                // Don't throw an exception for beta internals
+        boolean isBetaEdition = com.ibm.ws.kernel.productinfo.ProductInfo.getBetaEdition();
+        if (isBetaEdition && !!!issuedBetaMessage) {
+            Tr.info(tc, "BETA: BELL SPI Visibility and BELL Properties has been invoked by class " + this.getClass().getName() + " for the first time.");
+            issuedBetaMessage = true;
+        }
+        return isBetaEdition;
+    }
+
+    @SuppressWarnings("restriction")
+    boolean getSpiVisibility(boolean enableSpiVisibility, String libraryId) {
+        if (enableSpiVisibility) {
```

Comment:
Yes.  I verified the config service sets the default value for attribute `enableSpiVisibility` when the attribute is not configured within the BELL XML and when the attribute is configured incorrectly (i.e. not configured as `true` or `false`). The feature runtime further safeguards against this concern by relying on Java to unbox the Boolean value to boolean when passing the parameter by value to method `getSpiVisibility()`.
 
 I'll add a test for invalid `enableSpiVisibility` configuration to the BELL FAT which checks for this condition:
`CWWKG0083W: A validation failure occurred while processing the [enableSpiVisibility] property, value = [XXX]. Default value in use: [false].`

Reply:
> The feature runtime further safeguards against this concern by relying on Java to unbox the Boolean value to boolean when passing the parameter by value to method `getSpiVisibility()`.

That was what triggered my concern.  Because unboxing a null `Boolean` will result in a NPE (which is hard to debug when it does happen because the stacktrace doesn't make it obvious on Java < 17).  But if we are guaranteed to not have a null here then it should be fine.

Reply:
Oh no -- exactly the opposite of what I intended!  I hardened and clarified the solution against NPE.  
```
boolean getSpiVisibility(Boolean enableSpiVisibility, String libraryId) {
    boolean spiVisibility = Boolean.TRUE.equals(enableSpiVisibility);
    if (spiVisibility) {
        ...
```